### PR TITLE
CVSRP-4582b Updated routing for March Rti/Regular Length page

### PR DESCRIPTION
**Pull request by mywyau**

### CODE REVIEW

**Comment by mywyau** [comment]
branch made from CVSRP-4575

File: it/controllers/RegularLengthEmployedControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          httpStatus(OK),
+          titleOf(regularLengthEmployed)
+        )
+      }
+    }
+  }
+
+  "POST /regular-length-employed" when {
+
+    "enters a No valid answer" when {
```
Comment (LewisMitchell) [inline it/controllers/RegularLengthEmployedControllerISpec.scala:46]: Could there also be a test for disabling the feature switch and it routing to the pay date page? Also a test for when the user enters 'Yes' and it also routes to the pay date page regardless of the feature switch state? As well as some kind of sad path (a bad request with an invalid form passed in)?

File: test/navigation/NavigatorSpecWithApplication.scala [hunk 2]
```diff
@@ -519,95 +619,151 @@ class NavigatorSpecWithApplication extends SpecBaseControllerSpecs with CoreTest
...
-            .withClaimPeriodStart("2020,11,1")
-            .withClaimPeriodEnd("2020,11,8")
-            .withEmployeeStartDate("2019, 3, 30")
-            .withPayDate(List("2020,10,31", "2020,11,7"))
-            .withLastPayDate("2020,11,7")
-            .withFurloughStartDate("2020,11,1")
-        ) mustBe routes.LastYearPayController.onPageLoad(1)
+        "the ExtensionTwoNewStarterFlow switch is disabled" should {
+
+          "go to correct next page" in {
```
Comment (mywyau) [inline test/navigation/NavigatorSpecWithApplication.scala:642]: old tests added back in

File: test/navigation/NavigatorSpecWithApplication.scala [hunk 3]
```diff
@@ -519,95 +619,151 @@ class NavigatorSpecWithApplication extends SpecBaseControllerSpecs with CoreTest
...
-            .withRtiSubmission(EmployeeRTISubmission.No)
-        ) mustBe routes.PreviousFurloughPeriodsController.onPageLoad()
+        "the ExtensionTwoNewStarterFlow switch is enabled" should {
+
+          "return the OnPayrollBefore30thOct2020 page" in {
+
+            enable(ExtensionTwoNewStarterFlow)
+
+            navigator.nextPage(
+              EmployeeRTISubmissionPage,
```
Comment (LewisMitchell) [inline test/navigation/NavigatorSpecWithApplication.scala:733]: **Could there be a test for when the user enters 'Yes' on this page and we ensure that the existing routing logic won't change?** The design says there will be a new page but we can just ensure that this routing still exists for now as we can change it later?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, need to run the AT's and waiting for test result. Just a few minor test changes. 😄

File: it/controllers/RegularLengthEmployedControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            .withClaimPeriodStart("2020, 11, 1")
+            .withClaimPeriodEnd("2020, 11, 30")
+            .withFurloughStartDate("2020, 11, 1")
+            .withFurloughStatus()
+            .withPaymentFrequency(Monthly)
+            .withNiCategory()
+            .withPensionStatus()
+            .withPayMethod()
+            .withLastPayDate("2020, 10, 31")
+            .withPayDate(List("2020, 10, 31"))
```
Comment (mywyau) [inline it/controllers/RegularLengthEmployedControllerISpec.scala:50]: note: I am assuming these are the mandatory answers needed just adjusted the claim start dates and stuff needed for this pages nav logic

Comment (mywyau) [reply it/controllers/RegularLengthEmployedControllerISpec.scala:50]: they have a method which builds monthly mandatory answers just based it off that
`  def mandatoryAnswersOnRegularMonthly =
    emptyUserAnswers
      .withClaimPeriodStart("2020, 3, 1")
      .withClaimPeriodEnd("2020, 3, 31")
      .withFurloughStartDate("2020, 3, 1")
      .withFurloughStatus()
      .withPaymentFrequency(Monthly)
      .withNiCategory()
      .withPensionStatus()
      .withPayMethod()
      .withLastPayDate("2020, 3, 31")
      .withPayDate(List("2020, 2, 29", "2020, 3, 31"))`

**Pull request merged**